packages/core/src/middleware/actions.ts

```diff
@@ -45,16 +61,9 @@ export const middlewareActions =
   ): { middleware: ClientMiddleware } => ({
     middleware: {
       customMiddleware: overrides.customMiddleware ?? noopMiddleware,
-      dummyPaymasterAndData: overrides.paymasterAndData?.dummyPaymasterAndData
-        ? async (struct, { account }) =>
-            account.getEntryPoint().version === "0.6.0"
-              ? {
-                  ...struct,
-                  paymasterAndData:
-                    overrides.paymasterAndData!.dummyPaymasterAndData(),
-                }
-              : struct
-        : defaultPaymasterAndData,
```

Comment:
This was fine as it was and just needed support for `0.7.0`

```ts
async (struct, { account }) =>
            account.getEntryPoint().version === "0.6.0"
              ? {
                  ...struct,
                  paymasterAndData:
                    overrides.paymasterAndData!.dummyPaymasterAndData(),
                }
              : {
                 paymaster: // take the first 20 bytes of overrides.paymasterAndData!.dummyPaymasterAndData(),
                 paymasterData: // take the remaining bytes
               }
```

Reply:
paymasterData? I thought we only set 'paymaster' field. @dphilipson do we need to set dummy paymasterData as well for gas estimation and paymaster query?

Reply:
@moldy530 okay so just parse from dummyPaymasterAndData returned Hex as Address?

Reply:
discussed here https://alchemyinsights.slack.com/archives/C0669LA6JQH/p1714084846367119

and updated with the aligned approach